assets/src/blocks/Submenu/SubmenuEditor.js

```diff
@@ -0,0 +1,123 @@
+import { Fragment } from '@wordpress/element';
+import { Button, PanelBody } from '@wordpress/components';
+import { SubmenuLevel } from './SubmenuLevel';
+import { SubmenuItems } from './SubmenuItems';
+import { InspectorControls } from '@wordpress/block-editor';
+import { getSubmenuStyle } from './getSubmenuStyle';
+import { makeHierarchical } from './makeHierarchical';
+import { getHeadingsFromBlocks} from './getHeadingsFromBlocks';
+import { useSelect } from '@wordpress/data';
+import { deepClone } from '../../functions/deepClone';
+
+const { __ } = wp.i18n;
+const { RichText } = wp.blockEditor;
+
+const renderEdit = (attributes, setAttributes) => {
+  function addLevel() {
+    const [previousLastLevel] = attributes.levels.slice(-1);
+    const newLevel = previousLastLevel.heading + 1;
+    setAttributes({ levels: attributes.levels.concat({ heading: newLevel, link: false, style: 'none' }) });
+  }
+
+  function onHeadingChange(index, value) {
+    const levels = deepClone(attributes.levels);
+    levels[index].heading = Number(value);
+    setAttributes({ levels });
+  }
+
+  function onLinkChange(index, value) {
+    const levels = deepClone(attributes.levels);
+    levels[index].link = value;
+    setAttributes({ levels });
+  }
+
+  function onStyleChange(index, value) {
+    const levels = deepClone(attributes.levels);
+    levels[index].style = value; // Possible values: "none", "bullet", "number"
+    setAttributes({ levels });
+  }
+
+  function removeLevel() {
+    setAttributes({ levels: attributes.levels.slice(0, -1) });
+  }
+
+  function getMinLevel(attributes, index) {
+    if (index === 0) {
+      return null;
+    }
+
+    return attributes.levels[index-1].heading;
+  }
+
+  return (
+    <InspectorControls>
+      <PanelBody title={__('Setting', 'planet4-blocks-backend')}>
+        {attributes.levels.map((level, i) => (
+          <SubmenuLevel
+            {...level}
+            onHeadingChange={onHeadingChange}
+            onLinkChange={onLinkChange}
+            onStyleChange={onStyleChange}
+            index={i}
+            key={i}
+            minLevel={getMinLevel(attributes, i)}
+          />
+        ))}
+        <Button
+          isPrimary
+          onClick={addLevel}
+          disabled={attributes.levels.length >= 3 || attributes.levels.slice(-1)[0].heading === 0}
+          style={{ marginRight: 5 }}
+        >
+          {__('Add level', 'planet4-blocks-backend')}
+        </Button>
+        <Button
+          isSecondary
+          onClick={removeLevel}
+          disabled={attributes.levels.length <= 1}
+        >
+          {__('Remove level', 'planet4-blocks-backend')}
+        </Button>
+      </PanelBody>
+    </InspectorControls>
+  );
+}
+
+const renderView = (attributes, setAttributes, className) => {
+
+  const blocks = useSelect(select => select('core/editor').getBlocks(), null);
```

Comment:
`select('core/editor').getBlocks()` is deprecated (since https://github.com/WordPress/gutenberg/pull/15770, there's a message in console), we have to use `select('core/block-editor').getBlocks()`